### PR TITLE
Delta Storage: initial project setup

### PR DESCRIPTION
**Pull request by scottsand-db**
Move over the `LogStore` and `CloseableIterator` interfaces from `delta-core`'s `io.delta.storage` package

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -158,6 +159,18 @@ lazy val contribs = (project in file("contribs"))
...
 
+// TODO javastyle tests
+// TODO unidoc
+lazy val storage = (project in file("storage"))
+  .settings (
+    name := "delta-storage",
+    commonSettings,
+    releaseSettings,
+    libraryDependencies ++= Seq(
+      "org.apache.hadoop" % "hadoop-client-api" % "3.3.1" % "provided"
```
Comment (tdas) [inline build.sbt:170]: why this version of hadoop? is there any restriction for this? 
if so put that in comments.

File: build.sbt [hunk 2]
```diff
@@ -158,6 +159,19 @@ lazy val contribs = (project in file("contribs"))
...
 
+// TODO javastyle tests
+// TODO unidoc
+lazy val storage = (project in file("storage"))
+  .settings (
+    name := "delta-storage",
+    commonSettings,
+    releaseSettings,
+    libraryDependencies ++= Seq(
+      // User can provide any version >= 3.1.0 (released in 2018). We don't use any new fancy APIs.
```
Comment (zsxwing) [inline build.sbt:170]: We should be able to support any 2.x or 3.x version.

**Review by zsxwing — COMMENTED** [review]
Do you plan to change the artifact name to remove Scala version and make the release process only releases one jar in this PR? Or just plan to do that later?

File: build.sbt (hunk 2 above)
Comment (scottsand-db) [reply build.sbt:170]: do you want me to update the comment or update the actual dependency?

**Comment by scottsand-db** [comment]
@zsxwing 
> Do you plan to change the artifact name to remove Scala version and make the release process only releases one jar in this PR? Or just plan to do that later?

I plan to do this later. I will add a TODO comment. this will let me focus more on core development, so that we can start creating OSS issues and hopefully get some community help. is this plan okay?

File: build.sbt (hunk 2 above)
Comment (zsxwing) [reply build.sbt:170]: just update the comment

**Comment by zsxwing** [comment]
> I plan to do this later. I will add a TODO comment. this will let me focus more on core development, so that we can start creating OSS issues and hopefully get some community help. is this plan okay?

SGTM

**Review by zsxwing — APPROVED** [review]
LGTM

**Pull request merged**